Framework/DataHandling/src/LoadNexusMonitors2.cpp

```diff
@@ -491,53 +500,37 @@ void LoadNexusMonitors2::splitMutiPeriodHistrogramData(
   }
 
   // Y array should be divisible by the number of periods
-  if (m_workspace->blocksize() % numPeriods != 0) {
+  if (m_multiPeriodCounts[0].size() % numPeriods != 0) {
     g_log.warning()
         << "Attempted to split multiperiod histogram workspace with "
-        << m_workspace->blocksize() << "data entries, into " << numPeriods
+        << m_multiPeriodCounts[0].size() << "data entries, into " << numPeriods
         << "periods."
            " Aborted.\n";
     return;
   }
 
   WorkspaceGroup_sptr wsGroup(new WorkspaceGroup);
-  size_t yLength = m_workspace->blocksize() / numPeriods;
+  size_t yLength = m_multiPeriodCounts[0].size() / numPeriods;
   size_t xLength = yLength + 1;
   size_t numSpectra = m_workspace->getNumberHistograms();
   ISISRunLogs monLogCreator(m_workspace->run());
+
+  BinEdges edges = m_multiPeriodBinEdges[0];
+
   for (size_t i = 0; i < numPeriods; i++) {
     // create the period workspace
     API::MatrixWorkspace_sptr wsPeriod =
         API::WorkspaceFactory::Instance().create(m_workspace, numSpectra,
                                                  xLength, yLength);
 
-    // assign x values - restart at start for all periods
-    for (size_t wsIndex = 0; wsIndex < numSpectra; wsIndex++) {
-      MantidVec &outputVec = wsPeriod->dataX(wsIndex);
-      const MantidVec &inputVec = m_workspace->readX(wsIndex);
-      for (size_t index = 0; index < xLength; index++) {
-        outputVec[index] = inputVec[index];
-      }
-    }
+    auto offset = yLength * i;
 
-    // assign y values - use the values offset by the period number
     for (size_t wsIndex = 0; wsIndex < numSpectra; wsIndex++) {
-      MantidVec &outputVec = wsPeriod->dataY(wsIndex);
-      const MantidVec &inputVec = m_workspace->readY(wsIndex);
-      for (size_t index = 0; index < yLength; index++) {
-        outputVec[index] = inputVec[(yLength * i) + index];
-      }
-    }
+      auto inYBegin = m_multiPeriodCounts[wsIndex].cbegin() + offset;
```

Comment:
Before, `m_workspace` was used to store the multiperiod data which consisted of a number of histograms where the size of the y array would be at least twice the size of the x array. This violated the histogramdata type, so I stored the multiperiod data in `m_multiPeriodCounts/BinEdges`. The sizes of these `m_multiPeriodCounts` vectors correspond to the old size of `m_workspace->blocksize()`. The the calculation `m_multiPeriodCounts[0].size() % numPeriods` is identical. The code below is splitting each of the multiperiod histograms and storing them in the `wsPeriod` workspace which is designed to correctly store the multiperiod data. In essence, the workspace type was bastardized.